### ISSUE TITLE
datetime-min-max (DTZ901) is fooled by `tzinfo=None`

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ruff#28013]
### Summary

```python
from datetime import datetime

datetime.min.replace(tzinfo=None)                      # Missed
datetime.max.replace(tzinfo=None)                      # Missed
datetime.strptime("2026", "%Y").replace(tzinfo=None)    # DTZ007
```

The DTZ901 / datetime-min-max lint should alert that a naive datetime is being used here, but it doesn't. The analogous DTZ007 rule does do this correctly for the strptime call.

### Version

0.16.4
